dvc/utils/fs.py

```diff
@@ -53,3 +53,65 @@ def contains_symlink_up_to(path, base_path):
     if os.path.dirname(path) == path:
         return False
     return contains_symlink_up_to(os.path.dirname(path), base_path)
+
+
+class BaseTree(object):
```

Comment:
It feels like it is not the place for these classes. How about we move it to `dvc/repo/fs` or something?